src/main/java/com/fasterxml/jackson/core/io/schubfach/FloatToDecimal.java

```diff
@@ -405,14 +405,14 @@ private static int rop(long g, long cp) {
     /*
     Formats the decimal f 10^e.
      */
-    private int toChars(int f, int e) {
+    private int toChars(long f, int e) {
```

Comment:
I can't find any calls to this method where the input f is long - they are all int. I think there will be a performance penalty here as there will need to be an implicit cast from int to long on all calls to this method after this change.

Reply:
I guess the start is to provide a broken test case. I'd prefer not to deal with theory and more interested in actual test cases.